schema/typeMethods.go

```diff
@@ -66,6 +67,9 @@ func (t TypeEnum) RepresentationBehavior() datamodel.Kind {
 	// TODO: this should have a representation strategy switch too; sometimes that will indicate int representation behavior.
 	return datamodel.Kind_String
 }
+func (t TypeAny) RepresentationBehavior() datamodel.Kind {
+	return datamodel.Kind_Invalid // TODO: what can we possibly do here?
```

Comment:
Indeed.

Unions have a similar funny corner:

```
func (t TypeUnion) RepresentationBehavior() datamodel.Kind {
	switch t.representation.(type) {
	case UnionRepresentation_Kinded:
		return datamodel.Kind_Invalid // you can't know with this one, until you see the value (and thus can its inhabitant's behavior)!
```

I guess that if we were to dial types up to 11, this method should actually return a type that's `datamodel.Kind` plus another enum value for describing this case of "it depends".

Or we can keep (ab)using `Kind_Invalid` like this.

I don't know if we have any user complaints that provide a strong impetus one way or the other.

Reply:
can we have a `Kind_Unknown` for cases where we don't (yet) know?

Reply:
I think discussing this in a PR review isn't the best idea, and this isn't particuarly making things worse, so I've raised https://github.com/ipld/go-ipld-prime/issues/326. Marking as resolved for now :)